common/archiver/provider/provider.go

```diff
@@ -181,31 +208,19 @@ func (p *archiverProvider) GetVisibilityArchiver(scheme, serviceName string) (ar
 		return nil, ErrBootstrapContainerNotFound
 	}
 
-	var visibilityArchiver archiver.VisibilityArchiver
-	var err error
-
-	switch scheme {
-	case filestore.URIScheme:
-		if p.visibilityArchiverConfigs.Filestore == nil {
-			return nil, ErrArchiverConfigNotFound
-		}
-		visibilityArchiver, err = filestore.NewVisibilityArchiver(container, p.visibilityArchiverConfigs.Filestore)
-	case s3store.URIScheme:
-		if p.visibilityArchiverConfigs.S3store == nil {
-			return nil, ErrArchiverConfigNotFound
-		}
-		visibilityArchiver, err = s3store.NewVisibilityArchiver(container, p.visibilityArchiverConfigs.S3store)
-	case gcloud.URIScheme:
-		if p.visibilityArchiverConfigs.Gstorage == nil {
-			return nil, ErrArchiverConfigNotFound
-		}
-		visibilityArchiver, err = gcloud.NewVisibilityArchiver(container, p.visibilityArchiverConfigs.Gstorage)
-
-	default:
-		return nil, ErrUnknownScheme
+	constructor, ok := visibilityConstructors.Get(scheme)
+	if !ok {
+		return nil, fmt.Errorf("no visibility archiver constructor for scheme %q", scheme)
+	}
+
+	cfg, ok := p.historyArchiverConfigs[constructor.configKey]
```

Comment:
it should be visibility archiver

Reply:
ah, yea, I guess this got through builds because they have the same signature, and through tests because we rarely have different archivers... 🤔 

good catch, thanks!  will fix asap.